### DISCUSSION TITLE
Providing additional format and style to output information (using Debug.WriteLine)

### DISCUSSION THREAD

**Discussion by carloswm85** [viacheslav-lozinskyi/MetaOutput#24] [Q&A] [answered]
I was trying to follow your instruction for coloring the output text:

```console
Debug.WriteLine("FINISHED ✋ - Elapsed time: {0} @@@FOREGROUND RED", watch.Elapsed);
```

But it looks like I'm doing something wrong:

![image](https://github.com/viacheslav-lozinskyi/MetaOutput/assets/29048235/48551278-badb-403c-8191-c14773331ccc)

Help?

Also, what other format and style modifications could I add? And how?

**Comment by viacheslav-lozinskyi** [comment, marked as answer]
After receiving of your message, MetaOutput apply all known rules to that message.
That rule can override your settings.

In this situation rule is next: `if text message has substring "FINISHED" then foreground will be changed to green.`
This green color you see.

Each source has own set of rules.
In MetaOutput 2.0 that rules is impossible to change.
In MetaOutput 3.0 will be possible to change any existing rule and add / remove any number of own rules.
